File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        var n = s.Length;
        var i = 0; 
        var j = 0; 
        var maxSize = 0;
        
        HashSet<char> st = new HashSet<char>();
        
        if (s.Length == 0)
            return 0;
        
        while (i < n){
            if (!st.Contains(s[i])){
                st.Add(s[i]);
                maxSize = Math.Max(maxSize, (int)st.Count);
                ++i;
            }
            else{
                st.Remove(s[j]);
                ++j;
            }
        }
        return maxSize;
    }
}

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n = s.size(), i =0, j = 0;
        if (n == 0)
            return 0;
        set <char> st;
        int maxSize = 0;
        while (i < n){
            if(st.count(s[i]) == 0){
                st.insert(s[i]);
                maxSize = max(maxSize, (int)st.size());
                ++i;
            }
            else{
                st.erase(s[j]);
                j++;
            }
        }
        return maxSize;
    }
};